### ISSUE TITLE
Jobs with a survey spec do not launch in 16.0.0

### ISSUE THREAD

**Issue by rjosephwright** [ansible/awx#9058]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY

Jobs with a survey spec do not launch, browser console throws error.

##### ENVIRONMENT
* AWX version: 16.0.0
* AWX install method: docker on linux
* Ansible version:  Whatever is built into the 16.0.0 docker image
* Operating System: Debian 10
* Web Browser: Chrome 86.0.4240.198

##### STEPS TO REPRODUCE

Create a job template with a survey spec. In my case it was added with the Ansible `tower_job_template` module.

Click the launch icon to the right of the job name from the /templates screen.

##### EXPECTED RESULTS

Screen to launch the job should load.

##### ACTUAL RESULTS

A blank screen loads. The javascript console in Chrome shows the following error:

```
asyncToGenerator.js:6 Uncaught (in promise) TypeError: t.choices.split is not a function
    at gd (SurveyStep.jsx:109)
    at Xo (react-dom.production.min.js:153)
    at vs (react-dom.production.min.js:261)
    at cl (react-dom.production.min.js:246)
    at sl (react-dom.production.min.js:246)
    at Qs (react-dom.production.min.js:239)
    at react-dom.production.min.js:123
    at t.unstable_runWithPriority (scheduler.production.min.js:19)
    at Hi (react-dom.production.min.js:122)
    at Gi (react-dom.production.min.js:123)
```

This error seems to be located at https://github.com/ansible/awx/blob/16.0.0/awx/ui_next/src/components/LaunchPrompt/steps/SurveyStep.jsx#L109.

##### ADDITIONAL INFORMATION

Job templates which do not have a survey spec load when the launch button is clicked.

**Comment by mabashian** [comment]
@rjosephwright that error would indicate that you've got a multiple choice survey question with a `null` value for `choices`.  I was able to reproduce this same scenario locally against the head of the devel branch.  Can you confirm that that's what your survey spec looks like as well?  You can find this by using the api browser and navigating to `/api/v2/job_templates/:id_of_your_job_template/survey_spec/` or by looking at your browsers network tab when you attempt to launch the job.  You should see the request to GET the survey_spec in there.  I'm curious to see what the response looks like.

We'll fix this but if you've got a multiple choice question without any choices that doesn't really make sense.  You can probably fix this by updating that question's `choices` field to be an empty string (or even better a new line separated string of choices).

**Comment by rjosephwright** [comment]
@mabashian I've torn down the instance but I can spin one up later to get that output you requested.

However, I checked through all of my survey specs and I don't think there are any `multiplechoice` questions without `choices` defined.

For example:

```
    {
      "question_name": "Image project",
      "question_description": "....",
      "required": true,
      "type": "multiplechoice",
      "variable": "sap_source_image_project",
      "default": "rhel-sap-cloud",
      "choices": ["rhel-sap-cloud", "suse-sap-cloud"]
    },
```

I've been using the same survey specs successfully on 14.0 and didn't make any changes to them before trying them on 16.0.

**Comment by rjosephwright** [comment]
Just to verify, I did a quick iteration over all spec files and determined that all `multiplechoice` questions do have `choices` defined.

```
In [35]: for specfile in specfiles: 
    ...:     with open(specfile) as f: 
    ...:         j = json.load(f) 
    ...:     print([q for q in j['spec'] if q['type'] == 'multiplechoice' and not q.get('choices')]) 
    ...:                                                                                                                                                                     
[]
[]
[]
[]
[]
[]
[]
[]
[]
[]
[]
```

**Comment by areguera** [comment]
In the UI, I started to receive a message like the following on a job template that was working without any issue in 15.1.0 before the update to 16.0.0:

![image](https://user-images.githubusercontent.com/5214519/104633964-b8217300-567e-11eb-95ba-b45b35121e4b.png)

This message blocks the job template execution in the preview step. The launch button is disabled.

**Comment by jakemcdermott** [comment]
"It's fixed" - john

**Comment by shaaga** [comment]
@jakemcdermott can you please share what is the fix? I am facing the same issue in AWX 17.1.0 with all job templates having a multiplechoice field, irrespective of whether it has choices defined or choices field is null?
